src/EnergyPlus/SwimmingPool.cc

```diff
@@ -460,6 +416,59 @@ namespace SwimmingPool {
         }
     }
 
+    void SwimmingPoolData::ErrorCheckSetupPoolSurface(std::string const Alpha1,
+                                                      std::string const Alpha2,
+                                                      std::string const cAlphaField2,
+                                                      bool &ErrorsFound
+    )
+    {
+    
+        static std::string const RoutineName("ErrorCheckSetupPoolSurface: "); // include trailing blank space
+        static std::string const CurrentModuleObject("SwimmingPool:Indoor");
+        
+        if (this->SurfacePtr <= 0) {
+            ShowSevereError(RoutineName + "Invalid " + cAlphaField2 + " = " + Alpha2);
+            ShowContinueError("Occurs in " + CurrentModuleObject + " = " + Alpha1);
+            ErrorsFound = true;
+        } else if (DataSurfaces::Surface(this->SurfacePtr).IsRadSurfOrVentSlabOrPool) {
```

Comment:
This is really a great addition.  

While the reduced lookups have the potential for a miniscule speedup in this case (it can when it is applied more widespread: https://github.com/NREL/EnergyPlus/pull/4192#issuecomment-39848360), more importantly it reduces the amount of code for a developer to process.  

I would suggest in the future taking it one step farther and creating a reference to the surface of interest.  You'll want to first check if the surface <= 0 and leave early if it is:
```
if (this->SurfacePtr <= 0) {
    ShowSevereError(RoutineName + "Invalid " + cAlphaField2 + " = " + Alpha2);
    ShowContinueError("Occurs in " + CurrentModuleObject + " = " + Alpha1);
    ErrorsFound = true;
    return;
}
```

Then after this point, you can comfortably get a reference to a surface:

```
auto & surface = DataSurfaces::Surface(this->SurfacePtr);
```

You can then replace all the instances of `DataSurfaces::Surface(this->SurfacePtr)` with `surface`.  Once again, this isn't really a performance thing, but more a cognitive complexity overhead.  When a developer reads this code, there are still a bunch of array lookups going on amidst all the actual logic and functionality.  Cleaning up the code in ways like this can make the function's intention much more clear.